tests/virtualization/vagrant/boxes/tumbleweed.pm

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test for vagrant and packaged addons
+# Maintainer: dancermak <dcermak@suse.com>
+
+use strict;
+use warnings;
+use base "basetest";
+use testapi;
+use utils;
+use File::Basename;
+
+
+sub run() {
+    select_console('root-console');
+    zypper_call('in ansible');
+
+    select_console('user-console');
+
+    # version = Tumbleweed, Leap 15 etc
+    my $version = get_required_var('VERSION');
+    my $arch    = get_required_var('ARCH');
+    # FIXME: don't hardcode the build in production
+    # my $build = get_required_var('BUILD');
+    my $build = "20190721";
+
+    # This is a constant defined in the KIWI config file, we don't expect it to change
+    my $boxversion = '1.0';
+
+    my %boxes = (
+        # openSUSE-Tumbleweed-Vagrant.x86_64-1.0-{libvirt|virtualbox}-Snapshot20190704.vagrant.{libvirt|virtualbox}.box
+        libvirt    => "openSUSE-$version-Vagrant.$arch-$boxversion-libvirt-Snapshot$build.vagrant.libvirt.box",
+        virtualbox => "openSUSE-$version-Vagrant.$arch-$boxversion-virtualbox-Snapshot$build.vagrant.virtualbox.box"
+    );
```

Comment:
this and the following lines look like the equivalent to simply

```perl
my @providers = qw(libvirt virtualbox);
foreach my $provider (@providers) {
    my $boxname = "openSUSE-$version-Vagrant.$arch-$boxversion-$provider-Snapshot$build.vagrant.$provider.box";
    …
```

Reply:
Well, I'm now reusing the boxname later on in a second loop and find the current solution a little cleaner.